compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/extended/CanBeReplacedWithOperatorAssignmentChecker.kt

```diff
@@ -22,16 +22,19 @@ import org.jetbrains.kotlin.fir.types.coneType
 import org.jetbrains.kotlin.lexer.KtTokens
 import org.jetbrains.kotlin.psi.KtBinaryExpression
 import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtOperationReferenceExpression
 
 object CanBeReplacedWithOperatorAssignmentChecker : FirExpressionChecker<FirVariableAssignment>() {
-    override fun check(functionCall: FirVariableAssignment, context: CheckerContext, reporter: DiagnosticReporter) {
-        val lValue = functionCall.lValue
+    override fun check(assignment: FirVariableAssignment, context: CheckerContext, reporter: DiagnosticReporter) {
+        val lValue = assignment.lValue
         if (lValue !is FirResolvedNamedReference) return
-        val operator = functionCall.psi?.children?.getOrNull(1) ?: return
-        if (operator.text != "=") return
+
+        val operator = assignment.psi?.children?.getOrNull(1) ?: return
```

Comment:
This is bad, because that checker won't work in light tree mode. I see, that this code was written before, so it's ok to push it in this PR, but you should fix it in future

Reply:
Fix for this issue will be presented in special PR (I will add support for light tree in all extended checkers in it)